File: iOS/swiftui-journey-module/JourneyModuleSample/JourneyModuleSample/Callbacks/BooleanAttributeInputCallbackView.swift
//
//  BooleanAttributeInputCallbackView.swift
//  JourneyModuleSample
//
//  Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
//
//  This software may be modified and distributed under the terms
//  of the MIT license. See the LICENSE file for details.
//

import SwiftUI
import PingJourney

/**
 * A SwiftUI view for capturing boolean (true/false) attribute values during authentication flows.
 *
 * This view displays a toggle switch with a descriptive prompt label, allowing users to enable
 * or disable a specific attribute or setting. Common use cases include opt-in preferences,
 * feature flags, or binary configuration options during registration or profile updates.
 *
 * **User Action Required:** YES - User must toggle the switch to set their preference.
 *
 * The UI displays the prompt text on the left with a toggle switch on the right. The toggle
 * reflects the current value and updates the callback immediately when changed.
 */
struct BooleanAttributeInputCallbackView: View {
    let callback: BooleanAttributeInputCallback
    let onNodeUpdated: () -> Void

    @State var value: Bool = false

    var body: some View {
        HStack {
            Text(callback.prompt)
                .foregroundColor(.primary)

            Spacer()

            Toggle("", isOn: $value)
                .toggleStyle(SwitchToggleStyle())
                .onChange(of: value) { newValue in
                    callback.value = newValue
                }
        }
        .padding()
        .onAppear {
            value = callback.value
        }
    }
}

File: flutter/flutter-journey/lib/ui/features/journey/views/widgets/callbacks/boolean_attribute_input_callback_view.dart
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import 'package:flutter/material.dart';
import 'package:ping_journey/ping_journey.dart';

class BooleanAttributeInputCallbackView extends StatelessWidget {
  const BooleanAttributeInputCallbackView({
    super.key,
    required this.callback,
    required this.onChanged,
  });

  final BooleanAttributeInputCallback callback;
  final VoidCallback onChanged;

  @override
  Widget build(BuildContext context) {
    final label = callback.prompt ?? callback.name;
    return SwitchListTile(
      contentPadding: EdgeInsets.zero,
      title: Text(callback.required ? '$label *' : label),
      value: callback.value,
      onChanged: (value) {
        callback.value = value;
        onChanged();
      },
    );
  }
}
